vividus/src/main/java/org/vividus/bdd/expression/HashExpressionProcessor.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2019-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.bdd.expression;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.inject.Named;
+
+import org.apache.commons.codec.digest.DigestUtils;
+import org.apache.commons.lang3.EnumUtils;
+
+@Named
+public class HashExpressionProcessor extends AbstractExpressionProcessor<String>
+{
+    private static final Pattern HASH_PATTERN = Pattern.compile("^calculateHash\\((.*), (.*)\\)$",
+            Pattern.CASE_INSENSITIVE);
+
+    protected HashExpressionProcessor()
+    {
+        super(HASH_PATTERN);
+    }
+
+    @Override
+    protected String evaluateExpression(Matcher expressionMatcher)
+    {
+        String hash;
+        String algorithm = expressionMatcher.group(1);
+        String originalString = expressionMatcher.group(2);
+        HashAlgorithmType hashAlgorithmType = EnumUtils.getEnumIgnoreCase(HashAlgorithmType.class,
+                algorithm.replace("-", ""), HashAlgorithmType.UNKNOWN);
+        switch (hashAlgorithmType)
+        {
+            case MD2:
+                hash = DigestUtils.md2Hex(originalString);
+                break;
+            case MD5:
+                hash = DigestUtils.md5Hex(originalString);
+                break;
+            case SHA1:
+                hash = DigestUtils.sha1Hex(originalString);
+                break;
+            case SHA256:
+                hash = DigestUtils.sha256Hex(originalString);
+                break;
+            case SHA384:
+                hash = DigestUtils.sha384Hex(originalString);
+                break;
+            case SHA512:
+                hash = DigestUtils.sha512Hex(originalString);
+                break;
```

Comment:
move switch logic to enum, for example: https://github.com/vividus-framework/vividus/blob/master/vividus-bdd-engine/src/main/java/org/vividus/bdd/steps/StringComparisonRule.java